homeassistant/components/tado/__init__.py

```diff
@@ -107,6 +114,17 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
+    async def add_meter_reading(call: ServiceCall) -> None:
+        """Send meter reading to Tado."""
+        _LOGGER.info("Service call to add_meter_reading")
+        await hass.async_add_executor_job(
+            tadoconnector.set_meter_reading, call.data[ATTR_READING]
+        )
+
+    hass.services.async_register(
```

Comment:
it is no longer allowed to register services in `async_setup_entry`, but only in `async_setup`

Reply:
This init file does not (yet) include an `async_setup` method. Can this be resovled with a simple cut and paste, leaving the rest of the `async_setup_entry` as it was?

Reply:
no, I'm afraid it's not gonna be that easy... You should leave the rest of the initialization where it is, but only move the service init to `async_setup`. The problem which you are gonna face is, that the `entry` is not available in there, so it's gonna be a bit more work to get the `entry.data` to init the `tadoconnector`. I'd suggest you take a look at the `lametric` or `easyenergy` components to see how they handle this.

Reply:
> it is no longer allowed to register services in `async_setup_entry`, but only in `async_setup`

Please resolve conversation at your discretion.